src/Voron/Platform/Posix/PosixAbstractPager.cs

```diff
@@ -47,5 +49,24 @@ protected internal override unsafe void PrefetchRanges(Win32MemoryMapNativeMetho
         protected PosixAbstractPager(StorageEnvironmentOptions options, bool canPrefetchAhead, bool usePageProtection = false) : base(options, canPrefetchAhead, usePageProtection: usePageProtection)
         {
         }
+        
+        protected override void DisposeInternal()
+        {
+            if (_fd != -1)
+            {
+                // note that the orders of operations is important here, we first unlink the file
+                // we are supposed to be the only one using it, so Linux would be ready to delete it
+                // and hopefully when we close it, won't waste any time trying to sync the memory state
+                // to disk just to discard it
+                if (DeleteOnClose)
```

Comment:
Need `madvise` here

Reply:
madvise should be used in the unmapping events, and maybe we can just fadvise instead : fadvise(fd, DONTNEED) ?

Reply:
okey, irrelevant. fadvise is not good here. 
We will:
if (DeleteOnClose == true)
  madvise(addr, DONTNEED);
munmap(addr);